component/src/main/java/org/wso2/rule/validator/document/Document.java

```diff
@@ -0,0 +1,249 @@
+import org.wso2.rule.validator.ruleset.Rule;
+import org.wso2.rule.validator.ruleset.RuleThen;
+import org.wso2.rule.validator.ruleset.Ruleset;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Document class to represent a target document for rules to be applied.
+ */
+public class Document {
+
+    private String documentString;
+    private Object document;
+    Format format;
+
+    public Document(String documentString) {
+        LoadSettings settings = LoadSettings.builder().build();
+        Load yamlLoader = new Load(settings);
+        Object yamlData = yamlLoader.loadFromString(documentString);
+
+        if (yamlData == null) {
+            return;
+        }
+
+        Gson gson = new GsonBuilder().setPrettyPrinting().serializeNulls().create();
+        this.documentString = gson.toJson(yamlData);
+
+        this.document = JsonPath.parse(this.documentString).json();
+        resolveReferences();
+
+        // Read format
+        Map<String, Object> documentMap = (Map<String, Object>) yamlData;
+        if (documentMap.containsKey(Constants.OPENAPI_KEY)) {
+            String oasVersion = (String) documentMap.get(Constants.OPENAPI_KEY);
+            if (oasVersion.startsWith(Constants.OAS_3_1_VERSION)) {
+                this.format = Format.OAS3_1;
+            } else if (oasVersion.startsWith(Constants.OAS_3_0_VERSION)) {
+                this.format = Format.OAS3_0;
+            } else {
+                this.format = Format.OAS3;
+            }
+        } else if (documentMap.containsKey(Constants.SWAGGER_KEY)) {
+            this.format = Format.OAS2;
+        }
+    }
+
+    public ArrayList<FunctionResult> lint(Ruleset ruleset) {
+        // TODO: Add parsing errors to the result set
+
+        // TODO: Filter enabled and relevant rules
+
+        ArrayList<FunctionResult> results = new ArrayList<>();
+
+        for (Rule rule : ruleset.rules.values()) {
+            for (String given : rule.given) {
+                // TODO: Implement aliases
+                if (given.startsWith(Constants.ALIAS_PREFIX)) {
+                    continue;
+                }
+                try {
+                    Configuration config = Configuration.builder().options(Option.AS_PATH_LIST).build();
+                    List<String> paths = JsonPath.using(config).parse(this.document).read(given);
+                    for (String path : paths) {
+                        results.addAll(lintNode(path, rule));
+                    }
+                    // log("Json Path resolved: " + given);
+                } catch (PathNotFoundException e) {
+                    // log("Json Path not found: " + given);
+                } catch (InvalidPathException e) {
+                    // log("Unsupported Json Path: " + given);
+                    // TODO: Implement json path plus features
```

Comment:
Shall we handle the errors properly here in the catch blocks?

Reply:
These exception are meant to be ignored since the flow must not break during the rule validation. It can only break during the ruleset validation phase

Reply:
Removed the logs with https://github.com/wso2/rule-validator/pull/23